lua/nvim-tree/explorer/filters.lua

```diff
@@ -73,7 +73,20 @@ end
 ---@param path string
 ---@param bookmarks table<string, boolean> absolute paths bookmarked
 local function bookmark(path, bookmarks)
-  return M.config.filter_no_bookmark and not bookmarks[path]
+  if not M.config.filter_no_bookmark then
+    return false
+  end
+
+  -- add trailing slash to make it match only mark's parent directory
+  -- not it's siblings
+  local dict = utils.path_add_trailing(path)
```

Comment:
`dict` does not mean anything.

You can mutate the argument e.g. `path = utils.path_add_trailing(path)`. That's an accepted lua idiom.

Reply:
```lua
  -- add trailing slash to make it match only mark's parent directory
  -- not it's siblings
  local parent = utils.path_add_trailing(path)
  for mark, _ in pairs(bookmarks) do
    if path == mark or vim.fn.stridx(mark, parent) == 0 then
      return false
    end
  end
```

I used `path` (unmodified) for matching with mark itself with 
```lua 
path == mark  -- stop here, no need to run match string
```

`dict` (now is `parent` for clear understand) is to avoid matching with mark's siblings 
for example:
```
/foo/bar/bookmark_me (marked)
/foo/bar/book  (this will show up)
```

We can simply avoid it by adding a trailing slash to `path`
```lua
local parent = utils.path_add_trailing(path)
```

Now mark's siblings can't match 
```
/foo/bar/bookmark_me (marked)
/foo/bar/book/
```